2017-12-01 09:24:30,372 - snapshots - ERROR - ====================
2017-12-01 13:53:23.928086
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 09:24:30,372 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-01 09:26:00,398 - snapshots - ERROR - ====================
2017-12-01 14:24:30.372818
Get request failed for: https://www.bitstamp.net/api/order_book/


2017-12-01 09:26:00,403 - snapshots - ERROR - HTTPSConnectionPool(host='www.bitstamp.net', port=443): Max retries exceeded with url: /api/order_book/ (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x117fa9690>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
2017-12-01 10:01:38,975 - snapshots - ERROR - ====================
2017-12-01 14:30:33.744499
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 10:01:38,975 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-01 10:44:15,865 - snapshots - ERROR - ====================
2017-12-01 15:31:48.839770
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 10:44:15,866 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-01 10:45:45,898 - snapshots - ERROR - ====================
2017-12-01 15:44:15.866632
Get request failed for: https://www.bitstamp.net/api/order_book/


2017-12-01 10:45:45,899 - snapshots - ERROR - HTTPSConnectionPool(host='www.bitstamp.net', port=443): Max retries exceeded with url: /api/order_book/ (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x117fa9bd0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
2017-12-01 10:46:06,431 - db_util - ERROR - Failed to store in Spreads
2017-12-01 10:46:06,431 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: 'DESCRIBE `Spreads`']
2017-12-01 11:31:36,938 - snapshots - ERROR - ====================
2017-12-01 16:31:36.075852
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 11:31:36,939 - snapshots - ERROR - No JSON object could be decoded
2017-12-01 11:31:36,940 - snapshots - ERROR - Content: <!DOCTYPE html>
<!--[if lt IE 7]> <html class="no-js ie6 oldie" lang="en-US"> <![endif]-->
<!--[if IE 7]>    <html class="no-js ie7 oldie" lang="en-US"> <![endif]-->
<!--[if IE 8]>    <html class="no-js ie8 oldie" lang="en-US"> <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en-US"> <!--<![endif]-->
<head>
<meta http-equiv="refresh" content="0">

<meta http-equiv="set-cookie" content="cf_use_ob=443; expires=Fri, 01-Dec-17 16:28:47 GMT; path=/">

<meta http-equiv="set-cookie" content="cf_ob_info=504:3c676b2dde3e921e:EWR; expires=Fri, 01-Dec-17 16:28:47 GMT; path=/">


<title>api.gdax.com | 504: Gateway time-out</title>
<meta charset="UTF-8" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1" />
<meta name="robots" content="noindex, nofollow" />
<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1" />
<link rel="stylesheet" id="cf_styles-css" href="/cdn-cgi/styles/cf.errors.css" type="text/css" media="screen,projection" />
<!--[if lt IE 9]><link rel="stylesheet" id='cf_styles-ie-css' href="/cdn-cgi/styles/cf.errors.ie.css" type="text/css" media="screen,projection" /><![endif]-->
<style type="text/css">body{margin:0;padding:0}</style>
<!--[if lte IE 9]><script type="text/javascript" src="/cdn-cgi/scripts/jquery.min.js"></script><![endif]-->
<!--[if gte IE 10]><!--><script type="text/javascript" src="/cdn-cgi/scripts/zepto.min.js"></script><!--<![endif]-->
<script type="text/javascript" src="/cdn-cgi/scripts/cf.common.js"></script>

</head>
<body>
<div id="cf-wrapper">

    

    <div id="cf-error-details" class="cf-error-details-wrapper">
        <div class="cf-wrapper cf-error-overview">
            <h1>
              <span class="cf-error-type" data-translate="error">Error</span>
              <span class="cf-error-code">504</span>
              <small class="heading-ray-id">Ray ID: 3c676b2dde3e921e &bull; 2017-12-01 16:28:17 UTC</small>
            </h1>
            <h2 class="cf-subheadline" data-translate="error_desc">Gateway time-out</h2>
        </div><!-- /.error-overview -->

        <div class="cf-section cf-highlight cf-status-display">
            <div class="cf-wrapper">
                <div class="cf-columns cols-3">
                  
<div id="cf-browser-status" class="cf-column cf-status-item cf-browser-status ">
  <div class="cf-icon-error-container">
    <i class="cf-icon cf-icon-browser"></i>
    <i class="cf-icon-status cf-icon-ok"></i>
  </div>
  <span class="cf-status-desc" data-translate="browser_desc">You</span>
  <h3 class="cf-status-name" data-translate="browser_label">Browser</h3>
  <span class="cf-status-label" data-translate="browser_status_label">Working</span>
</div>

<div id="cf-cloudflare-status" class="cf-column cf-status-item cf-cloudflare-status ">
  <div class="cf-icon-error-container">
    <i class="cf-icon cf-icon-cloud"></i>
    <i class="cf-icon-status cf-icon-ok"></i>
  </div>
  <span class="cf-status-desc" data-translate="cloud_desc">Newark</span>
  <h3 class="cf-status-name" data-translate="cloud_label">Cloudflare</h3>
  <span class="cf-status-label" data-translate="cloud_status_label">Working</span>
</div>

<div id="cf-host-status" class="cf-column cf-status-item cf-host-status cf-error-source">
  <div class="cf-icon-error-container">
    <i class="cf-icon cf-icon-server"></i>
    <i class="cf-icon-status cf-icon-error"></i>
  </div>
  <span class="cf-status-desc" data-translate="server_desc">api.gdax.com</span>
  <h3 class="cf-status-name" data-translate="server_label">Host</h3>
  <span class="cf-status-label" data-translate="server_status_label">Error</span>
</div>

                </div>
            </div>
        </div><!-- /.status-display -->

        <div class="cf-section cf-wrapper">
            <div class="cf-columns two">
                <div class="cf-column">
                    <h2 data-translate="what_happened">What happened?</h2>
                    <p>The web server reported a gateway time-out error.</p>
                </div>

                <div class="cf-column">
                    <h2 data-translate="what_can_i_do">What can I do?</h2>
                    <p data-translate="try_again_in_a_few">Please try again in a few minutes.</p>
                </div>
            </div>
        </div><!-- /.section -->

        <div class="cf-error-footer cf-wrapper">
  <p>
    <span class="cf-footer-item">Cloudflare Ray ID: <strong>3c676b2dde3e921e</strong></span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span data-translate="your_ip">Your IP</span>: 67.80.186.88</span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span data-translate="performance_security_by">Performance &amp; security by</span> <a data-orig-proto="https" data-orig-ref="www.cloudflare.com/5xx-error-landing?utm_source=error_footer" id="brand_link" target="_blank">Cloudflare</a></span>
    
  </p>
</div><!-- /.error-footer -->


    </div><!-- /#cf-error-details -->
</div><!-- /#cf-wrapper -->

2017-12-01 11:37:31,629 - snapshots - ERROR - ====================
2017-12-01 16:37:26.529036
Get request failed for: https://www.bitstamp.net/api/order_book/


2017-12-01 11:37:31,630 - snapshots - ERROR - HTTPSConnectionPool(host='www.bitstamp.net', port=443): Read timed out.
2017-12-01 15:14:40,880 - db_util - ERROR - Failed to store in Spreads
2017-12-01 15:14:40,881 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: 'DESCRIBE `Spreads`']
2017-12-01 15:41:23,810 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 15:41:23,811 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10579.87', 'bid', u'16.23154268', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10579', 'bid', u'0.025', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10578.98', 'bid', u'0.02142202', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10578.95', 'bid', u'0.04', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10578.94', 'bid', u'1.11923818', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10578.9', 'bid', u'0.125', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10578.87', 'bid', u'0.017466', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10578', 'bid', u'4.12082735', '4452642803')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10599.5', 'ask', u'2.68825165', '4452642803'), (datetime.datetime(2017, 12, 1, 20, 14, 41, 892403), u'10599.54', 'ask', u'0.010025', '4452642803'))]
2017-12-01 16:01:15,840 - snapshots - ERROR - ====================
2017-12-01 20:42:02.686512
Get request failed for: https://www.bitstamp.net/api/order_book/


2017-12-01 16:01:15,841 - snapshots - ERROR - HTTPSConnectionPool(host='www.bitstamp.net', port=443): Read timed out. (read timeout=5)
2017-12-01 16:01:33,848 - db_util - ERROR - Failed to store in Spreads
2017-12-01 16:01:33,848 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: 'DESCRIBE `Spreads`']
2017-12-01 16:37:09,117 - db_util - ERROR - Failed to store in Spreads
2017-12-01 16:37:09,515 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10800.00', u'10799.99', u'10799.99', u'10768.01', u'10768', 4452825459, datetime.datetime(2017, 12, 1, 21, 1, 57, 964724))]
2017-12-01 16:39:44,400 - snapshots - ERROR - ====================
2017-12-01 21:39:39.301117
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 16:39:44,401 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-01 16:39:51,075 - snapshots - ERROR - ====================
2017-12-01 21:39:45.973884
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 16:39:51,075 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-01 17:14:25,085 - snapshots - ERROR - ====================
2017-12-01 22:14:25.072042
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 17:14:25,588 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Max retries exceeded with url: /products/BTC-USD/book?level=2 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x117f65890>: Failed to establish a new connection: [Errno 50] Network is down',))
2017-12-01 20:58:31,484 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 20:58:31,485 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: 'DESCRIBE `GdaxSnapshots`']
2017-12-01 21:37:14,275 - db_util - ERROR - Failed to store in Spreads
2017-12-01 21:37:14,280 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: 'SHOW FULL TABLES FROM `bitcoin_db`']
2017-12-01 21:54:54,111 - snapshots - ERROR - ====================
2017-12-02 02:54:53.895745
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 21:54:54,112 - snapshots - ERROR - No JSON object could be decoded
2017-12-01 21:54:54,112 - snapshots - ERROR - Content: <!DOCTYPE html>
<!--[if lt IE 7]> <html class="no-js ie6 oldie" lang="en-US"> <![endif]-->
<!--[if IE 7]>    <html class="no-js ie7 oldie" lang="en-US"> <![endif]-->
<!--[if IE 8]>    <html class="no-js ie8 oldie" lang="en-US"> <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en-US"> <!--<![endif]-->
<head>
<meta http-equiv="refresh" content="0">

<meta http-equiv="set-cookie" content="cf_use_ob=443; expires=Sat, 02-Dec-17 02:52:01 GMT; path=/">

<meta http-equiv="set-cookie" content="cf_ob_info=504:3c6afc207c240efd:EWR; expires=Sat, 02-Dec-17 02:52:01 GMT; path=/">


<title>api.gdax.com | 504: Gateway time-out</title>
<meta charset="UTF-8" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1" />
<meta name="robots" content="noindex, nofollow" />
<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1" />
<link rel="stylesheet" id="cf_styles-css" href="/cdn-cgi/styles/cf.errors.css" type="text/css" media="screen,projection" />
<!--[if lt IE 9]><link rel="stylesheet" id='cf_styles-ie-css' href="/cdn-cgi/styles/cf.errors.ie.css" type="text/css" media="screen,projection" /><![endif]-->
<style type="text/css">body{margin:0;padding:0}</style>
<!--[if lte IE 9]><script type="text/javascript" src="/cdn-cgi/scripts/jquery.min.js"></script><![endif]-->
<!--[if gte IE 10]><!--><script type="text/javascript" src="/cdn-cgi/scripts/zepto.min.js"></script><!--<![endif]-->
<script type="text/javascript" src="/cdn-cgi/scripts/cf.common.js"></script>

</head>
<body>
<div id="cf-wrapper">

    

    <div id="cf-error-details" class="cf-error-details-wrapper">
        <div class="cf-wrapper cf-error-overview">
            <h1>
              <span class="cf-error-type" data-translate="error">Error</span>
              <span class="cf-error-code">504</span>
              <small class="heading-ray-id">Ray ID: 3c6afc207c240efd &bull; 2017-12-02 02:51:31 UTC</small>
            </h1>
            <h2 class="cf-subheadline" data-translate="error_desc">Gateway time-out</h2>
        </div><!-- /.error-overview -->

        <div class="cf-section cf-highlight cf-status-display">
            <div class="cf-wrapper">
                <div class="cf-columns cols-3">
                  
<div id="cf-browser-status" class="cf-column cf-status-item cf-browser-status ">
  <div class="cf-icon-error-container">
    <i class="cf-icon cf-icon-browser"></i>
    <i class="cf-icon-status cf-icon-ok"></i>
  </div>
  <span class="cf-status-desc" data-translate="browser_desc">You</span>
  <h3 class="cf-status-name" data-translate="browser_label">Browser</h3>
  <span class="cf-status-label" data-translate="browser_status_label">Working</span>
</div>

<div id="cf-cloudflare-status" class="cf-column cf-status-item cf-cloudflare-status ">
  <div class="cf-icon-error-container">
    <i class="cf-icon cf-icon-cloud"></i>
    <i class="cf-icon-status cf-icon-ok"></i>
  </div>
  <span class="cf-status-desc" data-translate="cloud_desc">Newark</span>
  <h3 class="cf-status-name" data-translate="cloud_label">Cloudflare</h3>
  <span class="cf-status-label" data-translate="cloud_status_label">Working</span>
</div>

<div id="cf-host-status" class="cf-column cf-status-item cf-host-status cf-error-source">
  <div class="cf-icon-error-container">
    <i class="cf-icon cf-icon-server"></i>
    <i class="cf-icon-status cf-icon-error"></i>
  </div>
  <span class="cf-status-desc" data-translate="server_desc">api.gdax.com</span>
  <h3 class="cf-status-name" data-translate="server_label">Host</h3>
  <span class="cf-status-label" data-translate="server_status_label">Error</span>
</div>

                </div>
            </div>
        </div><!-- /.status-display -->

        <div class="cf-section cf-wrapper">
            <div class="cf-columns two">
                <div class="cf-column">
                    <h2 data-translate="what_happened">What happened?</h2>
                    <p>The web server reported a gateway time-out error.</p>
                </div>

                <div class="cf-column">
                    <h2 data-translate="what_can_i_do">What can I do?</h2>
                    <p data-translate="try_again_in_a_few">Please try again in a few minutes.</p>
                </div>
            </div>
        </div><!-- /.section -->

        <div class="cf-error-footer cf-wrapper">
  <p>
    <span class="cf-footer-item">Cloudflare Ray ID: <strong>3c6afc207c240efd</strong></span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span data-translate="your_ip">Your IP</span>: 67.80.186.88</span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span data-translate="performance_security_by">Performance &amp; security by</span> <a data-orig-proto="https" data-orig-ref="www.cloudflare.com/5xx-error-landing?utm_source=error_footer" id="brand_link" target="_blank">Cloudflare</a></span>
    
  </p>
</div><!-- /.error-footer -->


    </div><!-- /#cf-error-details -->
</div><!-- /#cf-wrapper -->

2017-12-01 22:37:36,474 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:37:36,475 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:36-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10999.99', 'bid', u'15.261123', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10999.84', 'bid', u'0.01', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10999.03', 'bid', u'2.01', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10999', 'bid', u'0.025', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10998.99', 'bid', u'0.015004', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10998.89', 'bid', u'0.01364387', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10998.78', 'bid', u'0.01285351', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10992.68', 'bid', u'0.01', '4454230926')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'11027.7', 'ask', u'0.13564353', '4454230926'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'11029', 'ask', u'0.109', '4454230926'))]
2017-12-01 22:37:36,823 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:37:36,824 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:36-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10939.33', 'bid', u'0.06107641', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10938.91', 'bid', u'0.01800000', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10926.29', 'bid', u'0.20300000', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10926.28', 'bid', u'5.98510000', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10926.24', 'bid', u'0.20000000', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10920.73', 'bid', u'11.00000000', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10916.24', 'bid', u'0.20000000', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'10914.31', 'bid', u'0.00684906', '1512185847')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'11049.00', 'ask', u'0.01000000', '1512185847'), (datetime.datetime(2017, 12, 2, 3, 37, 36, 111451), u'11049.44', 'ask', u'50.00000000', '1512185847'))]
2017-12-01 22:37:36,940 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:37:36,940 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:36' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.32', u'10939.33', u'10939.33', u'11000', u'10999.99', 4454230926, datetime.datetime(2017, 12, 2, 3, 37, 36, 111451))]
2017-12-01 22:37:38,283 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:37:38,283 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:38-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10999.99', 'bid', u'15.73876959', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10999.84', 'bid', u'0.01', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10999.03', 'bid', u'2.01', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10999', 'bid', u'0.025', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10998.99', 'bid', u'0.015004', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10998.89', 'bid', u'0.01364387', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10998.78', 'bid', u'0.01285351', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10992.68', 'bid', u'0.01', '4454230974')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'11029.99', 'ask', u'0.01', '4454230974'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'11030', 'ask', u'0.04', '4454230974'))]
2017-12-01 22:37:39,284 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:37:39,285 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:38-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10939.38', 'bid', u'0.06381120', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10939.33', 'bid', u'0.49000000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10938.91', 'bid', u'0.01800000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10938.83', 'bid', u'0.46900000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10938.33', 'bid', u'0.47600000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10937.83', 'bid', u'0.47000000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10936.87', 'bid', u'0.20300000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'10936.86', 'bid', u'0.20300000', '1512185856')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'11049.00', 'ask', u'0.01000000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 37, 950923), u'11049.44', 'ask', u'50.00000000', '1512185856'))]
2017-12-01 22:37:39,393 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:37:39,393 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:38' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.32', u'10939.38', u'10939.38', u'11000', u'10999.99', 4454230974, datetime.datetime(2017, 12, 2, 3, 37, 37, 950923))]
2017-12-01 22:37:40,739 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:37:40,739 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:40-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10999.99', 'bid', u'16.39552024', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10999.84', 'bid', u'0.01', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10999.03', 'bid', u'2.01', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10999', 'bid', u'0.025', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10998.99', 'bid', u'0.015004', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10998.89', 'bid', u'0.01364387', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10998.78', 'bid', u'0.01285351', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10992.68', 'bid', u'0.01', '4454231084')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'11029', 'ask', u'0.109', '4454231084'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'11029.55', 'ask', u'3.1', '4454231084'))]
2017-12-01 22:37:41,058 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:37:41,059 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:40-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10939.38', 'bid', u'0.06381120', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10939.33', 'bid', u'0.49000000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10938.91', 'bid', u'0.01800000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10938.83', 'bid', u'0.46900000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10938.33', 'bid', u'0.47600000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10937.83', 'bid', u'0.47000000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10936.87', 'bid', u'0.20300000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'10936.86', 'bid', u'0.20300000', '1512185856')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'11049.00', 'ask', u'0.01000000', '1512185856'), (datetime.datetime(2017, 12, 2, 3, 37, 40, 402563), u'11049.44', 'ask', u'50.00000000', '1512185856'))]
2017-12-01 22:37:41,167 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:37:41,167 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:37:40' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.32', u'10939.38', u'10939.38', u'11000', u'10999.99', 4454231084, datetime.datetime(2017, 12, 2, 3, 37, 40, 402563))]
2017-12-01 22:38:00,628 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:00,629 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:00-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10999.99', 'bid', u'16.04286717', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10999.9', 'bid', u'0.159', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10999.84', 'bid', u'0.01', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10999.03', 'bid', u'2.01', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10999', 'bid', u'0.025', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10998.99', 'bid', u'0.015004', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10998.89', 'bid', u'0.01364387', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10998.78', 'bid', u'0.01285351', '4454231819')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'11030.07', 'ask', u'0.26', '4454231819'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'11031.95', 'ask', u'0.01', '4454231819'))]
2017-12-01 22:38:00,976 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:00,976 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:00-10926.25' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10939.43', 'bid', u'0.90306631', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10938.91', 'bid', u'0.01800000', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10938.43', 'bid', u'0.47600000', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10937.93', 'bid', u'0.84000000', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10937.43', 'bid', u'0.46700000', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10936.93', 'bid', u'0.85300000', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10936.86', 'bid', u'0.20300000', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'10936.83', 'bid', u'4.71970000', '1512185874')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'11049.44', 'ask', u'50.00000000', '1512185874'), (datetime.datetime(2017, 12, 2, 3, 38, 0, 293801), u'11050.02', 'ask', u'0.01575456', '1512185874'))]
2017-12-01 22:38:01,101 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:01,101 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:00' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.32', u'10939.43', u'10939.43', u'11000', u'10999.99', 4454231819, datetime.datetime(2017, 12, 2, 3, 38, 0, 293801))]
2017-12-01 22:38:07,189 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:07,190 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:07-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10999.99', 'bid', u'16.85385202', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10999.9', 'bid', u'0.169', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10999.84', 'bid', u'0.01', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10999.03', 'bid', u'2.01', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10999', 'bid', u'0.475', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10998.99', 'bid', u'0.015004', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10998.89', 'bid', u'0.01364387', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10998.78', 'bid', u'0.01285351', '4454232150')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'11031.95', 'ask', u'0.01', '4454232150'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'11031.96', 'ask', u'0.02', '4454232150'))]
2017-12-01 22:38:07,567 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:07,567 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:07-10926.25' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10939.48', 'bid', u'0.06381062', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10939.44', 'bid', u'0.20000000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10938.91', 'bid', u'0.01800000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10936.84', 'bid', u'0.20300000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10936.83', 'bid', u'4.71970000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10926.52', 'bid', u'2.00000000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10926.41', 'bid', u'0.37660750', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'10926.25', 'bid', u'0.84980000', '1512185884')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'11049.00', 'ask', u'0.01000000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 6, 832491), u'11049.44', 'ask', u'50.00000000', '1512185884'))]
2017-12-01 22:38:07,693 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:07,693 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:07' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.32', u'10939.48', u'10939.48', u'11000', u'10999.99', 4454232150, datetime.datetime(2017, 12, 2, 3, 38, 6, 832491))]
2017-12-01 22:38:09,038 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:09,038 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:09-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10999.99', 'bid', u'16.98285202', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10999.9', 'bid', u'0.01', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10999.84', 'bid', u'0.01', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10999.03', 'bid', u'2.01', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10999', 'bid', u'0.475', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10998.99', 'bid', u'0.015004', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10998.89', 'bid', u'0.01364387', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10998.78', 'bid', u'0.01285351', '4454232263')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'11031.95', 'ask', u'0.01', '4454232263'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'11031.96', 'ask', u'0.02', '4454232263'))]
2017-12-01 22:38:09,365 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:09,366 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:09-10926.25' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10939.48', 'bid', u'0.06381062', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10939.44', 'bid', u'0.20000000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10938.91', 'bid', u'0.01800000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10936.84', 'bid', u'0.20300000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10936.83', 'bid', u'4.71970000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10926.52', 'bid', u'2.00000000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10926.41', 'bid', u'0.37660750', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'10926.25', 'bid', u'0.84980000', '1512185884')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'11049.00', 'ask', u'0.01000000', '1512185884'), (datetime.datetime(2017, 12, 2, 3, 38, 8, 698367), u'11049.44', 'ask', u'50.00000000', '1512185884'))]
2017-12-01 22:38:09,478 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:09,478 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:09' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.32', u'10939.48', u'10939.48', u'11000', u'10999.99', 4454232263, datetime.datetime(2017, 12, 2, 3, 38, 8, 698367))]
2017-12-01 22:38:19,287 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:19,287 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:19-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10999.99', 'bid', u'21.12644135', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10999.9', 'bid', u'0.01', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10999.84', 'bid', u'0.01', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10999.03', 'bid', u'2.01', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10999', 'bid', u'0.98861131', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10998.99', 'bid', u'0.015004', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10998.89', 'bid', u'0.01364387', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10998.78', 'bid', u'0.01285351', '4454232645')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'11029.99', 'ask', u'0.01', '4454232645'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'11030', 'ask', u'0.04', '4454232645'))]
2017-12-01 22:38:19,722 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:19,723 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:19-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10939.53', 'bid', u'0.06423859', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10939.49', 'bid', u'0.20000000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10938.91', 'bid', u'0.01800000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10936.84', 'bid', u'0.20300000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10936.83', 'bid', u'4.71970000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10926.28', 'bid', u'2.00000000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10926.25', 'bid', u'0.84980000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'10926.24', 'bid', u'0.57669050', '1512185893')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'11049.44', 'ask', u'50.00000000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 18, 946183), u'11050.02', 'ask', u'0.01575456', '1512185893'))]
2017-12-01 22:38:19,849 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:19,849 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:19' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10958.99', u'10939.53', u'10939.53', u'11000', u'10999.99', 4454232645, datetime.datetime(2017, 12, 2, 3, 38, 18, 946183))]
2017-12-01 22:38:21,189 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:21,191 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:21-10999.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10999.99', 'bid', u'21.34600972', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10999.9', 'bid', u'0.01', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10999.84', 'bid', u'0.01', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10999.03', 'bid', u'2.01', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10999', 'bid', u'0.98861131', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10998.99', 'bid', u'0.015004', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10998.89', 'bid', u'0.01364387', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10998.78', 'bid', u'0.01285351', '4454232706')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'11028.84', 'ask', u'0.05', '4454232706'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'11029', 'ask', u'0.109', '4454232706'))]
2017-12-01 22:38:21,519 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:21,519 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:21-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10939.53', 'bid', u'0.06423859', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10939.49', 'bid', u'0.20000000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10938.91', 'bid', u'0.01800000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10936.84', 'bid', u'0.20300000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10936.83', 'bid', u'4.71970000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10926.28', 'bid', u'2.00000000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10926.25', 'bid', u'0.84980000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'10926.24', 'bid', u'0.57669050', '1512185893')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'11049.44', 'ask', u'50.00000000', '1512185893'), (datetime.datetime(2017, 12, 2, 3, 38, 20, 860401), u'11050.02', 'ask', u'0.01575456', '1512185893'))]
2017-12-01 22:38:21,636 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:21,636 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:21' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10958.99', u'10939.53', u'10939.53', u'11000', u'10999.99', 4454232706, datetime.datetime(2017, 12, 2, 3, 38, 20, 860401))]
2017-12-01 22:38:22,959 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:22,960 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:23-11000.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'11000.01', 'bid', u'0.65892524', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'11000', 'bid', u'0.01', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10999.99', 'bid', u'20.42834493', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10999.9', 'bid', u'0.01', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10999.03', 'bid', u'0.01', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10999', 'bid', u'0.98861131', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10998.99', 'bid', u'0.015004', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10998.89', 'bid', u'0.01364387', '4454232831')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'11029.11', 'ask', u'2.78', '4454232831'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'11029.12', 'ask', u'3.1', '4454232831'))]
2017-12-01 22:38:24,144 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:24,144 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:23-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10939.53', 'bid', u'0.85523859', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10939.49', 'bid', u'0.20000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10939.03', 'bid', u'0.84000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10938.91', 'bid', u'0.01800000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10938.53', 'bid', u'0.84800000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10938.03', 'bid', u'0.84000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10937.53', 'bid', u'0.84500000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'10937.03', 'bid', u'0.85300000', '1512185901')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'11050.02', 'ask', u'0.01575456', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 22, 641934), u'11050.43', 'ask', u'0.02169101', '1512185901'))]
2017-12-01 22:38:24,258 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:24,258 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:23' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10958.99', u'10939.53', u'10939.53', u'11000.02', u'11000.01', 4454232831, datetime.datetime(2017, 12, 2, 3, 38, 22, 641934))]
2017-12-01 22:38:25,613 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:25,614 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:25-11003.86' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11004', 'bid', u'1.25353122', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11003.86', 'bid', u'0.01', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11001.09', 'bid', u'0.01364387', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11000', 'bid', u'0.06296541', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10999.99', 'bid', u'16.63384232', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10999.03', 'bid', u'0.01', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10999', 'bid', u'0.7028', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10998.78', 'bid', u'0.01285351', '4454233240')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11035.26', 'ask', u'0.1666', '4454233240'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11035.4', 'ask', u'2', '4454233240'))]
2017-12-01 22:38:25,939 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:25,939 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:25-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10939.53', 'bid', u'0.85523859', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10939.49', 'bid', u'0.20000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10939.03', 'bid', u'0.84000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10938.91', 'bid', u'0.01800000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10938.53', 'bid', u'0.84800000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10938.03', 'bid', u'0.84000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10937.53', 'bid', u'0.84500000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'10937.03', 'bid', u'0.85300000', '1512185901')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11050.02', 'ask', u'0.01575456', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 25, 260018), u'11050.43', 'ask', u'0.02169101', '1512185901'))]
2017-12-01 22:38:26,054 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:26,054 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:25' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10958.99', u'10939.53', u'10939.53', u'11004.01', u'11004', 4454233240, datetime.datetime(2017, 12, 2, 3, 38, 25, 260018))]
2017-12-01 22:38:27,388 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:27,389 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:27-11003.86' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11004', 'bid', u'6.38940221', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11003.86', 'bid', u'0.49092297', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11001.09', 'bid', u'0.01364387', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11000', 'bid', u'0.05296541', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10999.99', 'bid', u'16.34563202', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10999.03', 'bid', u'0.01', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10999', 'bid', u'0.7028', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10998.78', 'bid', u'0.01285351', '4454233298')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11032.59', 'ask', u'8.8', '4454233298'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11035.26', 'ask', u'0.1666', '4454233298'))]
2017-12-01 22:38:27,719 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:27,719 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:27-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10939.53', 'bid', u'0.85523859', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10939.49', 'bid', u'0.20000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10939.03', 'bid', u'0.84000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10938.91', 'bid', u'0.01800000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10938.53', 'bid', u'0.84800000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10938.03', 'bid', u'0.84000000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10937.53', 'bid', u'0.84500000', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'10937.03', 'bid', u'0.85300000', '1512185901')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11050.02', 'ask', u'0.01575456', '1512185901'), (datetime.datetime(2017, 12, 2, 3, 38, 27, 59660), u'11050.43', 'ask', u'0.02169101', '1512185901'))]
2017-12-01 22:38:27,822 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:27,822 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:27' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10958.99', u'10939.53', u'10939.53', u'11004.01', u'11004', 4454233298, datetime.datetime(2017, 12, 2, 3, 38, 27, 59660))]
2017-12-01 22:38:29,155 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:29,155 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:29-11003.86' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11004', 'bid', u'7.72574253', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11003.86', 'bid', u'0.49092297', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11001.09', 'bid', u'0.01364387', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11000', 'bid', u'2.40296541', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10999.99', 'bid', u'16.33464717', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10999.03', 'bid', u'0.01', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10999', 'bid', u'0.7028', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10998.78', 'bid', u'0.01285351', '4454233381')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11032.58', 'ask', u'1.32', '4454233381'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11032.59', 'ask', u'8.8', '4454233381'))]
2017-12-01 22:38:30,246 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:30,246 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:29-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10939.54', 'bid', u'0.49900000', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10939.53', 'bid', u'0.84100000', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10939.03', 'bid', u'0.46900000', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10938.53', 'bid', u'0.84800000', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10938.03', 'bid', u'0.84000000', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10937.53', 'bid', u'0.84500000', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10937.03', 'bid', u'0.85300000', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'10936.85', 'bid', u'0.29900000', '1512185908')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11050.43', 'ask', u'0.02169101', '1512185908'), (datetime.datetime(2017, 12, 2, 3, 38, 28, 824439), u'11050.57', 'ask', u'0.00072306', '1512185908'))]
2017-12-01 22:38:30,366 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:30,367 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:29' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.13', u'10939.54', u'10939.54', u'11004.01', u'11004', 4454233381, datetime.datetime(2017, 12, 2, 3, 38, 28, 824439))]
2017-12-01 22:38:39,180 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:39,181 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:39-11004.01' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11004.66', 'bid', u'10.71240918', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11004.56', 'bid', u'0.01', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11004.52', 'bid', u'0.01', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11004.01', 'bid', u'2', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11004', 'bid', u'0.02295141', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11003.66', 'bid', u'0.014997', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11003.55', 'bid', u'0.01364387', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11000.66', 'bid', u'0.01', '4454233874')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11043.42', 'ask', u'3', '4454233874'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11044', 'ask', u'0.6', '4454233874'))]
2017-12-01 22:38:39,511 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:39,511 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:39-10949.51' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10949.51', 'bid', u'0.52070000', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10949.01', 'bid', u'0.46900000', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10948.51', 'bid', u'0.84700000', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10948.01', 'bid', u'0.83900000', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10947.51', 'bid', u'0.84400000', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10947.01', 'bid', u'0.46800000', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10939.54', 'bid', u'0.20000000', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'10937.04', 'bid', u'0.38500000', '1512185914')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11050.57', 'ask', u'0.00072306', '1512185914'), (datetime.datetime(2017, 12, 2, 3, 38, 38, 813049), u'11051.06', 'ask', u'1.19777649', '1512185914'))]
2017-12-01 22:38:39,630 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:39,630 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:39' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10949.51', u'10949.51', u'11004.67', u'11004.66', 4454233874, datetime.datetime(2017, 12, 2, 3, 38, 38, 813049))]
2017-12-01 22:38:52,033 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:38:52,033 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:52-11000.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11008.13', 'bid', u'4.54498605', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11007.76', 'bid', u'0.016', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11007.02', 'bid', u'0.01364387', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11004.99', 'bid', u'0.19113167', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11004.67', 'bid', u'0.6', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11004.66', 'bid', u'0.08852204', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11004.55', 'bid', u'0.96841969', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11004', 'bid', u'2.31375475', '4454234842')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11038.77', 'ask', u'0.12', '4454234842'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11039.96', 'ask', u'0.01', '4454234842'))]
2017-12-01 22:38:52,373 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:38:52,373 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:52-10926.25' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10949.56', 'bid', u'0.90553094', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10949.52', 'bid', u'0.20000000', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10949.51', 'bid', u'0.03170000', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10949.06', 'bid', u'0.84000000', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10948.56', 'bid', u'0.47500000', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10948.06', 'bid', u'0.83900000', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10947.56', 'bid', u'0.84400000', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'10947.06', 'bid', u'0.85200000', '1512185926')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11050.02', 'ask', u'0.01575456', '1512185926'), (datetime.datetime(2017, 12, 2, 3, 38, 51, 680772), u'11050.43', 'ask', u'0.02169101', '1512185926'))]
2017-12-01 22:38:52,483 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:38:52,484 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:38:52' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.31', u'10949.56', u'10949.56', u'11008.14', u'11008.13', 4454234842, datetime.datetime(2017, 12, 2, 3, 38, 51, 680772))]
2017-12-01 22:39:02,498 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:02,499 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:02-11000.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11008.13', 'bid', u'6.57693951', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11008', 'bid', u'0.1', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11007.98', 'bid', u'0.01', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11007.76', 'bid', u'0.016', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11007.5', 'bid', u'0.5', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11007.02', 'bid', u'0.01364387', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11004.99', 'bid', u'0.011', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11004.66', 'bid', u'0.03996038', '4454235230')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11038.74', 'ask', u'8.46849323', '4454235230'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11038.75', 'ask', u'2.55', '4454235230'))]
2017-12-01 22:39:03,820 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:03,820 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:02-10926.26' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10949.57', 'bid', u'0.20000000', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10949.51', 'bid', u'0.03170000', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10940.64', 'bid', u'3.98970000', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10939.54', 'bid', u'0.13071158', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10930.17', 'bid', u'12.23000000', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10926.26', 'bid', u'0.67545420', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10926.25', 'bid', u'0.84980000', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'10926.24', 'bid', u'0.20000000', '1512185939')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11050.57', 'ask', u'0.00072306', '1512185939'), (datetime.datetime(2017, 12, 2, 3, 39, 2, 168732), u'11051.06', 'ask', u'1.19777649', '1512185939'))]
2017-12-01 22:39:03,950 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:03,950 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:02' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10949.57', u'10949.57', u'11008.14', u'11008.13', 4454235230, datetime.datetime(2017, 12, 2, 3, 39, 2, 168732))]
2017-12-01 22:39:12,134 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:12,134 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:12-11008.14' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11008.22', 'bid', u'6.86105982', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11008.21', 'bid', u'0.3', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11008.19', 'bid', u'1.19300113', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11008.15', 'bid', u'0.97810299', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11008.14', 'bid', u'1.45', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11008.13', 'bid', u'2.05293019', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11008.07', 'bid', u'0.01', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11007.76', 'bid', u'0.016', '4454235926')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11038.74', 'ask', u'8.46849323', '4454235926'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11038.77', 'ask', u'0.12', '4454235926'))]
2017-12-01 22:39:12,458 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:12,458 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:12-10926.26' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10956.31', 'bid', u'0.05085013', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10949.57', 'bid', u'0.20000000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10949.51', 'bid', u'0.03170000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10944.99', 'bid', u'0.29900000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10944.98', 'bid', u'3.50620000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10940.68', 'bid', u'0.44900000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10940.67', 'bid', u'2.00000000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'10939.54', 'bid', u'0.13071158', '1512185948')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11051.06', 'ask', u'1.19777649', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 11, 797171), u'11052.02', 'ask', u'0.01575456', '1512185948'))]
2017-12-01 22:39:12,569 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:12,569 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:12' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.31', u'10956.31', u'11008.23', u'11008.22', 4454235926, datetime.datetime(2017, 12, 2, 3, 39, 11, 797171))]
2017-12-01 22:39:13,934 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:13,934 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:14-11008.14' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11008.22', 'bid', u'6.88495828', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11008.19', 'bid', u'1.03', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11008.15', 'bid', u'0.97810299', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11008.14', 'bid', u'1.45', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11008.13', 'bid', u'2.03293019', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11008.07', 'bid', u'0.01', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11007.76', 'bid', u'0.016', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11007.5', 'bid', u'0.5', '4454236015')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11038.77', 'ask', u'0.12', '4454236015'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11039.93', 'ask', u'2', '4454236015'))]
2017-12-01 22:39:14,257 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:14,258 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:14-10926.26' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10956.31', 'bid', u'0.05085013', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10949.57', 'bid', u'0.20000000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10949.51', 'bid', u'0.03170000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10944.99', 'bid', u'0.29900000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10944.98', 'bid', u'3.50620000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10940.68', 'bid', u'0.44900000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10940.67', 'bid', u'2.00000000', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'10939.54', 'bid', u'0.13071158', '1512185948')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11051.06', 'ask', u'1.19777649', '1512185948'), (datetime.datetime(2017, 12, 2, 3, 39, 13, 575241), u'11052.02', 'ask', u'0.01575456', '1512185948'))]
2017-12-01 22:39:14,367 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:14,367 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:14' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.31', u'10956.31', u'11008.23', u'11008.22', 4454236015, datetime.datetime(2017, 12, 2, 3, 39, 13, 575241))]
2017-12-01 22:39:18,120 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:18,121 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:18-11008.14' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.8', 'bid', u'1.06453649', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.7', 'bid', u'0.01', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.23', 'bid', u'1.8', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.22', 'bid', u'5.37349713', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.19', 'bid', u'0.43', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.15', 'bid', u'0.97810299', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.14', 'bid', u'0.25', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11008.13', 'bid', u'0.03293019', '4454236316')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11038.86', 'ask', u'0.05', '4454236316'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11039.93', 'ask', u'2', '4454236316'))]
2017-12-01 22:39:18,452 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:18,453 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:18-10926.26' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10956.36', 'bid', u'0.06549027', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10956.32', 'bid', u'0.20000000', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10956.31', 'bid', u'0.05085013', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10949.51', 'bid', u'0.03170000', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10945.01', 'bid', u'4.12810000', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10940.68', 'bid', u'0.30193651', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10940.67', 'bid', u'2.00000000', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'10939.54', 'bid', u'0.13071158', '1512185955')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11051.06', 'ask', u'1.19777649', '1512185955'), (datetime.datetime(2017, 12, 2, 3, 39, 17, 792939), u'11052.02', 'ask', u'0.01575456', '1512185955'))]
2017-12-01 22:39:18,577 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:18,577 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:18' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.36', u'10956.36', u'11008.81', u'11008.8', 4454236316, datetime.datetime(2017, 12, 2, 3, 39, 17, 792939))]
2017-12-01 22:39:28,400 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:28,400 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:28-11009.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11009.99', 'bid', u'0.59', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11009.98', 'bid', u'0.3', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11009.9', 'bid', u'0.09', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11009.88', 'bid', u'0.12', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11009.73', 'bid', u'0.01292182', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11009.23', 'bid', u'0.01283181', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11009', 'bid', u'0.014506', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11008.81', 'bid', u'0.6', '4454237716')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11042.56', 'ask', u'0.01', '4454237716'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11044', 'ask', u'0.6', '4454237716'))]
2017-12-01 22:39:29,384 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:29,385 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:28-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10956.41', 'bid', u'0.06548997', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10956.32', 'bid', u'0.20000000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10956.31', 'bid', u'0.05085013', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10949.51', 'bid', u'0.03170000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10940.80', 'bid', u'1.04993320', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10940.79', 'bid', u'2.00000000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10940.68', 'bid', u'4.18030000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'10939.55', 'bid', u'0.30413050', '1512185967')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11051.06', 'ask', u'1.19777649', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 28, 68893), u'11052.02', 'ask', u'0.01575456', '1512185967'))]
2017-12-01 22:39:29,528 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:29,529 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:28' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.41', u'10956.41', u'11010', u'11009.99', 4454237716, datetime.datetime(2017, 12, 2, 3, 39, 28, 68893))]
2017-12-01 22:39:32,806 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:32,806 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:32-11009.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11009.99', 'bid', u'5.37121485', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11009.98', 'bid', u'0.3', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11009.97', 'bid', u'0.27585005', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11009.73', 'bid', u'0.04856166', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11009.23', 'bid', u'0.01283181', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11009.01', 'bid', u'0.43035194', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11009', 'bid', u'0.014506', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11008.81', 'bid', u'0.6', '4454238126')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11044.03', 'ask', u'0.01297266', '4454238126'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11044.17', 'ask', u'8.46849469', '4454238126'))]
2017-12-01 22:39:33,184 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:33,184 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:32-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10956.41', 'bid', u'0.06548997', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10956.32', 'bid', u'0.20000000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10956.31', 'bid', u'0.05085013', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10949.51', 'bid', u'0.03170000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10940.80', 'bid', u'1.04993320', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10940.79', 'bid', u'2.00000000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10940.68', 'bid', u'4.18030000', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'10939.55', 'bid', u'0.30413050', '1512185967')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11051.06', 'ask', u'1.19777649', '1512185967'), (datetime.datetime(2017, 12, 2, 3, 39, 32, 455439), u'11052.02', 'ask', u'0.01575456', '1512185967'))]
2017-12-01 22:39:33,322 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:33,323 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:32' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.41', u'10956.41', u'11010', u'11009.99', 4454238126, datetime.datetime(2017, 12, 2, 3, 39, 32, 455439))]
2017-12-01 22:39:37,148 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:37,149 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:37-11009.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.61', 'bid', u'1.0122209', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.6', 'bid', u'0.01', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.59', 'bid', u'0.23153005', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.54', 'bid', u'0.0105059', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.29', 'bid', u'0.06688263', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.26', 'bid', u'0.04', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.24', 'bid', u'0.09', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11010.22', 'bid', u'0.09', '4454238803')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11046.97', 'ask', u'0.02062526', '4454238803'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11047.24', 'ask', u'0.05', '4454238803'))]
2017-12-01 22:39:37,498 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:37,499 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:37-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10956.42', 'bid', u'0.20000000', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10956.41', 'bid', u'0.48900000', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10956.31', 'bid', u'0.05085013', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10955.91', 'bid', u'0.84000000', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10955.41', 'bid', u'0.84600000', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10954.91', 'bid', u'0.83800000', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10954.41', 'bid', u'0.84400000', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'10953.91', 'bid', u'0.85200000', '1512185972')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11050.57', 'ask', u'0.00072306', '1512185972'), (datetime.datetime(2017, 12, 2, 3, 39, 36, 832573), u'11051.06', 'ask', u'1.19777649', '1512185972'))]
2017-12-01 22:39:37,600 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:37,601 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:37' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.42', u'10956.42', u'11012.18', u'11010.61', 4454238803, datetime.datetime(2017, 12, 2, 3, 39, 36, 832573))]
2017-12-01 22:39:40,942 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:40,943 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:41-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11012.17', 'bid', u'0.47678927', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11012.02', 'bid', u'0.01', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11010.74', 'bid', u'0.18', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11010.71', 'bid', u'0.03', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11010.67', 'bid', u'3', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11010.64', 'bid', u'0.27583326', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11010.59', 'bid', u'0.01196168', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11010.54', 'bid', u'0.0105059', '4454239284')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11044.18', 'ask', u'8.7', '4454239284'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11044.39', 'ask', u'1.18404', '4454239284'))]
2017-12-01 22:39:41,863 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:41,864 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:41-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10956.46', 'bid', u'0.06548967', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10956.42', 'bid', u'0.20000000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10956.31', 'bid', u'0.05085013', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10954.96', 'bid', u'0.36900000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10954.46', 'bid', u'0.37700000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10953.96', 'bid', u'0.38400000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10949.51', 'bid', u'0.03170000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'10947.56', 'bid', u'0.30174676', '1512185978')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11051.06', 'ask', u'1.19777649', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 40, 584060), u'11052.02', 'ask', u'0.01575456', '1512185978'))]
2017-12-01 22:39:41,987 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:41,987 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:41' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.46', u'10956.46', u'11012.18', u'11012.17', 4454239284, datetime.datetime(2017, 12, 2, 3, 39, 40, 584060))]
2017-12-01 22:39:43,343 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:43,343 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:43-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11014.99', 'bid', u'0.2272209', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11012.19', 'bid', u'0.02', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11012.17', 'bid', u'2.77687342', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11010.67', 'bid', u'2', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11010.59', 'bid', u'0.01196168', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11010.54', 'bid', u'0.0105059', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11010.31', 'bid', u'1.13070781', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11010', 'bid', u'2.5', '4454239475')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11044.44', 'ask', u'2', '4454239475'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11045', 'ask', u'0.11874173', '4454239475'))]
2017-12-01 22:39:43,674 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:43,674 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:43-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10956.46', 'bid', u'0.06548967', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10956.42', 'bid', u'0.20000000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10956.31', 'bid', u'0.05085013', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10954.96', 'bid', u'0.36900000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10954.46', 'bid', u'0.37700000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10953.96', 'bid', u'0.38400000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10949.51', 'bid', u'0.03170000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'10947.56', 'bid', u'0.30174676', '1512185978')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11051.06', 'ask', u'1.19777649', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 42, 990425), u'11052.02', 'ask', u'0.01575456', '1512185978'))]
2017-12-01 22:39:43,786 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:43,787 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:43' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.46', u'10956.46', u'11015', u'11014.99', 4454239475, datetime.datetime(2017, 12, 2, 3, 39, 42, 990425))]
2017-12-01 22:39:45,141 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:45,141 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:45-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11014.99', 'bid', u'0.47678927', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11014.84', 'bid', u'0.01', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11012.18', 'bid', u'0.6', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11012.17', 'bid', u'2.72292182', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11010.67', 'bid', u'2', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11010.59', 'bid', u'0.01196168', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11010.54', 'bid', u'0.0105059', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11010.31', 'bid', u'1.13070781', '4454239587')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11044.44', 'ask', u'2', '4454239587'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11045', 'ask', u'0.11874173', '4454239587'))]
2017-12-01 22:39:45,479 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:45,479 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:45-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10956.46', 'bid', u'0.06548967', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10956.42', 'bid', u'0.20000000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10956.31', 'bid', u'0.05085013', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10954.96', 'bid', u'0.36900000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10954.46', 'bid', u'0.37700000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10953.96', 'bid', u'0.38400000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10949.51', 'bid', u'0.03170000', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'10947.56', 'bid', u'0.30174676', '1512185978')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11051.06', 'ask', u'1.19777649', '1512185978'), (datetime.datetime(2017, 12, 2, 3, 39, 44, 791582), u'11052.02', 'ask', u'0.01575456', '1512185978'))]
2017-12-01 22:39:45,593 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:45,593 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:45' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10967.18', u'10956.46', u'10956.46', u'11015', u'11014.99', 4454239587, datetime.datetime(2017, 12, 2, 3, 39, 44, 791582))]
2017-12-01 22:39:56,943 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:39:56,943 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:57-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11014.99', 'bid', u'9.72014765', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11013.99', 'bid', u'0.014983', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11012.5', 'bid', u'0.5', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11010.59', 'bid', u'0.01196168', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11010.31', 'bid', u'1.13070781', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11010', 'bid', u'2.67', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11009.99', 'bid', u'0.01097466', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11009.98', 'bid', u'0.1', '4454240430')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11045', 'ask', u'0.11874173', '4454240430'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11046.97', 'ask', u'0.02062526', '4454240430'))]
2017-12-01 22:39:57,279 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:39:57,279 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:57-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10956.52', 'bid', u'0.06548932', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10956.47', 'bid', u'0.20000000', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10956.31', 'bid', u'0.05085013', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10954.97', 'bid', u'0.36900000', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10949.51', 'bid', u'0.03170000', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10939.61', 'bid', u'0.30402060', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10939.60', 'bid', u'5.91030000', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'10939.54', 'bid', u'0.13071158', '1512185991')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11050.43', 'ask', u'0.02169101', '1512185991'), (datetime.datetime(2017, 12, 2, 3, 39, 56, 605428), u'11050.57', 'ask', u'0.00072306', '1512185991'))]
2017-12-01 22:39:57,413 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:39:57,413 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:39:57' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10960.36', u'10956.52', u'10956.52', u'11015', u'11014.99', 4454240430, datetime.datetime(2017, 12, 2, 3, 39, 56, 605428))]
2017-12-01 22:40:05,268 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:05,268 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:05-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11014.99', 'bid', u'9.26000699', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11013.99', 'bid', u'0.024983', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11012.5', 'bid', u'0.5', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11010.59', 'bid', u'0.01196168', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11010', 'bid', u'2.67', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11009.99', 'bid', u'0.01097466', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11009.01', 'bid', u'0.43035194', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11008.8', 'bid', u'0.01097566', '4454240810')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11045', 'ask', u'0.11874173', '4454240810'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11046.97', 'ask', u'0.02062526', '4454240810'))]
2017-12-01 22:40:05,602 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:05,602 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:05-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10956.31', 'bid', u'0.84000000', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10955.81', 'bid', u'0.84000000', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10955.31', 'bid', u'0.84600000', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10954.81', 'bid', u'0.83800000', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10954.31', 'bid', u'0.84400000', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10953.81', 'bid', u'0.85200000', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10949.51', 'bid', u'0.03170000', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'10939.72', 'bid', u'2.00000000', '1512185998')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11050.43', 'ask', u'0.02169101', '1512185998'), (datetime.datetime(2017, 12, 2, 3, 40, 4, 933988), u'11050.57', 'ask', u'0.00072306', '1512185998'))]
2017-12-01 22:40:05,712 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:05,712 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:05' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10960.36', u'10956.31', u'10956.31', u'11015', u'11014.99', 4454240810, datetime.datetime(2017, 12, 2, 3, 40, 4, 933988))]
2017-12-01 22:40:23,043 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:23,043 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:23-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11014.99', 'bid', u'7.92856339', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11014.98', 'bid', u'0.0912133', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11014', 'bid', u'0.45', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11013.99', 'bid', u'0.024983', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11012.5', 'bid', u'0.5', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11010.59', 'bid', u'0.01196168', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11010', 'bid', u'2.68', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11009.99', 'bid', u'0.01097466', '4454241778')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11044.18', 'ask', u'8.7', '4454241778'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11044.39', 'ask', u'1.18404', '4454241778'))]
2017-12-01 22:40:23,372 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:23,373 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:23-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10956.31', 'bid', u'0.84000000', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10955.81', 'bid', u'0.84000000', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10955.31', 'bid', u'0.84600000', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10954.81', 'bid', u'0.83800000', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10954.31', 'bid', u'0.84400000', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10953.81', 'bid', u'0.85200000', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10949.51', 'bid', u'0.03170000', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'10948.53', 'bid', u'3.92050000', '1512186018')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11050.57', 'ask', u'0.00072306', '1512186018'), (datetime.datetime(2017, 12, 2, 3, 40, 22, 704576), u'11051.06', 'ask', u'1.19777649', '1512186018'))]
2017-12-01 22:40:23,482 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:23,482 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:23' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10960.36', u'10956.31', u'10956.31', u'11015', u'11014.99', 4454241778, datetime.datetime(2017, 12, 2, 3, 40, 22, 704576))]
2017-12-01 22:40:37,471 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:37,471 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:37-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11014.99', 'bid', u'8.12413755', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11014.98', 'bid', u'0.0912133', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11014.62', 'bid', u'0.02', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11014', 'bid', u'0.45', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11013.99', 'bid', u'0.014983', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11012.5', 'bid', u'0.5', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11010', 'bid', u'2.68', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11008.14', 'bid', u'0.25', '4454242392')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11044.39', 'ask', u'1.18404', '4454242392'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11045', 'ask', u'0.11874173', '4454242392'))]
2017-12-01 22:40:37,876 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:37,877 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:37-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10949.59', 'bid', u'1.13900000', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10949.58', 'bid', u'1.00000000', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10949.51', 'bid', u'0.03170000', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10949.09', 'bid', u'0.84000000', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10948.59', 'bid', u'0.84700000', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10948.53', 'bid', u'3.92050000', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10948.09', 'bid', u'0.83900000', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'10947.59', 'bid', u'0.84400000', '1512186031')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11050.43', 'ask', u'0.02169101', '1512186031'), (datetime.datetime(2017, 12, 2, 3, 40, 37, 109250), u'11050.57', 'ask', u'0.00072306', '1512186031'))]
2017-12-01 22:40:38,017 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:38,017 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:37' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10960.36', u'10949.59', u'10949.59', u'11015', u'11014.99', 4454242392, datetime.datetime(2017, 12, 2, 3, 40, 37, 109250))]
2017-12-01 22:40:39,382 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:39,382 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:39-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11014.99', 'bid', u'8.12413755', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11014.98', 'bid', u'0.0912133', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11014.89', 'bid', u'0.01', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11014.37', 'bid', u'0.02', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11014', 'bid', u'0.45', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11013.99', 'bid', u'0.014983', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11012.5', 'bid', u'0.5', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11010', 'bid', u'3.68', '4454242531')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11046.97', 'ask', u'0.02062526', '4454242531'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11047.24', 'ask', u'0.05', '4454242531'))]
2017-12-01 22:40:40,406 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:40,407 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:39-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10949.59', 'bid', u'0.29900000', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10949.58', 'bid', u'1.00000000', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10949.51', 'bid', u'0.03170000', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10948.53', 'bid', u'3.92050000', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10936.03', 'bid', u'2.00000000', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10936.00', 'bid', u'0.25000000', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10929.52', 'bid', u'11.00000000', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'10927.95', 'bid', u'0.00684628', '1512186038')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11050.43', 'ask', u'0.02169101', '1512186038'), (datetime.datetime(2017, 12, 2, 3, 40, 39, 22131), u'11050.57', 'ask', u'0.00072306', '1512186038'))]
2017-12-01 22:40:40,529 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:40,529 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:39' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10960.31', u'10949.59', u'10949.59', u'11015', u'11014.99', 4454242531, datetime.datetime(2017, 12, 2, 3, 40, 39, 22131))]
2017-12-01 22:40:48,587 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:48,587 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:48-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11014.99', 'bid', u'9.44342062', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11014.98', 'bid', u'0.0912133', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11014.89', 'bid', u'0.01', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11014.52', 'bid', u'0.02', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11014', 'bid', u'0.45', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11012.5', 'bid', u'0.5', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11010', 'bid', u'3.68', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11008.14', 'bid', u'0.25', '4454242987')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11044.18', 'ask', u'8.7', '4454242987'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11044.39', 'ask', u'1.18404', '4454242987'))]
2017-12-01 22:40:48,927 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:48,927 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:48-10926.25' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10949.60', 'bid', u'0.29900000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10949.59', 'bid', u'0.84000000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10949.58', 'bid', u'1.00000000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10949.51', 'bid', u'0.03170000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10949.09', 'bid', u'0.46900000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10948.53', 'bid', u'3.92050000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10948.09', 'bid', u'0.37000000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'10947.60', 'bid', u'0.44900000', '1512186041')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11050.02', 'ask', u'0.01575456', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 48, 245495), u'11050.43', 'ask', u'0.02169101', '1512186041'))]
2017-12-01 22:40:49,043 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:49,043 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:48' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10960.36', u'10949.60', u'10949.60', u'11015', u'11014.99', 4454242987, datetime.datetime(2017, 12, 2, 3, 40, 48, 245495))]
2017-12-01 22:40:50,367 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:50,367 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:50-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11014.99', 'bid', u'9.44342062', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11014.98', 'bid', u'0.0912133', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11014.89', 'bid', u'0.01', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11014.35', 'bid', u'0.02', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11014', 'bid', u'0.45', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11012.5', 'bid', u'0.5', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11010', 'bid', u'3.68', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11008.14', 'bid', u'0.25', '4454243077')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11045', 'ask', u'0.11874173', '4454243077'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11046.97', 'ask', u'0.02062526', '4454243077'))]
2017-12-01 22:40:50,681 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:50,683 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:50-10926.25' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10949.60', 'bid', u'0.29900000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10949.59', 'bid', u'0.84000000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10949.58', 'bid', u'1.00000000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10949.51', 'bid', u'0.03170000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10949.09', 'bid', u'0.46900000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10948.53', 'bid', u'3.92050000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10948.09', 'bid', u'0.37000000', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'10947.60', 'bid', u'0.44900000', '1512186041')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11050.02', 'ask', u'0.01575456', '1512186041'), (datetime.datetime(2017, 12, 2, 3, 40, 50, 54220), u'11050.43', 'ask', u'0.02169101', '1512186041'))]
2017-12-01 22:40:50,819 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:50,819 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:50' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10960.36', u'10949.60', u'10949.60', u'11015', u'11014.99', 4454243077, datetime.datetime(2017, 12, 2, 3, 40, 50, 54220))]
2017-12-01 22:40:52,159 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:52,159 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:52-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11014.99', 'bid', u'9.46439051', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11014.98', 'bid', u'0.0912133', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11014.89', 'bid', u'0.01', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11014', 'bid', u'0.45', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11013.99', 'bid', u'0.014983', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11012.5', 'bid', u'0.5', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11010', 'bid', u'3.68', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11008.14', 'bid', u'0.25', '4454243199')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11045', 'ask', u'0.11874173', '4454243199'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11046.97', 'ask', u'0.02062526', '4454243199'))]
2017-12-01 22:40:53,163 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:53,163 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:52-10926.25' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10949.59', 'bid', u'1.13900000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10949.58', 'bid', u'1.00000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10949.51', 'bid', u'0.03170000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10949.09', 'bid', u'0.46900000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10948.53', 'bid', u'3.92050000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10936.03', 'bid', u'2.00000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10936.00', 'bid', u'0.25000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'10929.53', 'bid', u'0.30420690', '1512186048')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11049.44', 'ask', u'50.00000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 51, 822139), u'11050.02', 'ask', u'0.01575456', '1512186048'))]
2017-12-01 22:40:53,299 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:53,299 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:52' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.31', u'10949.59', u'10949.59', u'11015', u'11014.99', 4454243199, datetime.datetime(2017, 12, 2, 3, 40, 51, 822139))]
2017-12-01 22:40:54,683 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-01 22:40:54,684 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:54-11010.00' for key 'PRIMARY'") [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11014.99', 'bid', u'9.45439051', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11014.85', 'bid', u'0.01', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11014.7', 'bid', u'0.01', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11014.5', 'bid', u'0.69', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11014', 'bid', u'0.45', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11013.99', 'bid', u'0.014983', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11012.5', 'bid', u'0.5', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11010', 'bid', u'3.68', '4454243310')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11045', 'ask', u'0.11874173', '4454243310'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11046.97', 'ask', u'0.02062526', '4454243310'))]
2017-12-01 22:40:55,008 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-01 22:40:55,008 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:54-10926.24' for key 'PRIMARY'") [SQL: u'INSERT INTO `BitstampSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10949.59', 'bid', u'1.13900000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10949.58', 'bid', u'1.00000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10949.51', 'bid', u'0.03170000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10949.09', 'bid', u'0.46900000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10948.53', 'bid', u'3.92050000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10936.03', 'bid', u'2.00000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10936.00', 'bid', u'0.25000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'10929.53', 'bid', u'0.30420690', '1512186048')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11049.44', 'ask', u'50.00000000', '1512186048'), (datetime.datetime(2017, 12, 2, 3, 40, 54, 302805), u'11050.02', 'ask', u'0.01575456', '1512186048'))]
2017-12-01 22:40:55,112 - db_util - ERROR - Failed to store in Spreads
2017-12-01 22:40:55,112 - db_util - ERROR - (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry '2017-12-02 03:40:54' for key 'PRIMARY'") [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10956.31', u'10949.59', u'10949.59', u'11015', u'11014.99', 4454243310, datetime.datetime(2017, 12, 2, 3, 40, 54, 302805))]
2017-12-01 22:43:11,526 - snapshots - ERROR - ====================
2017-12-02 03:43:06.427970
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 22:43:11,527 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-01 23:18:55,439 - snapshots - ERROR - ====================
2017-12-02 04:18:50.337302
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 23:18:55,440 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-01 23:19:06,615 - snapshots - ERROR - ====================
2017-12-02 04:19:01.529350
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-01 23:19:06,616 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Read timed out. (read timeout=5)
2017-12-02 03:56:50,283 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-02 03:56:50,285 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10879.98', 'bid', u'73.29414486', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10879.97', 'bid', u'0.1', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10879.02', 'bid', u'19', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10878.98', 'bid', u'1.7277825', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10878.9', 'bid', u'0.02', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10878.82', 'bid', u'0.01031409', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10878', 'bid', u'0.01', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10877', 'bid', u'0.091', '4454749866')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10899.5', 'ask', u'0.04359', '4454749866'), (datetime.datetime(2017, 12, 2, 5, 56, 18, 594510), u'10899.9', 'ask', u'0.0372', '4454749866'))]
2017-12-02 04:46:19,247 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-02 04:46:19,628 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: 'DESCRIBE `GdaxSnapshots`']
2017-12-02 04:47:49,663 - snapshots - ERROR - ====================
2017-12-02 09:46:19.639348
Get request failed for: https://www.bitstamp.net/api/order_book/


2017-12-02 04:47:49,663 - snapshots - ERROR - HTTPSConnectionPool(host='www.bitstamp.net', port=443): Max retries exceeded with url: /api/order_book/ (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x118806090>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
2017-12-02 06:04:28,307 - snapshots - ERROR - ====================
2017-12-02 11:02:58.277305
Get request failed for: https://www.bitstamp.net/api/order_book/


2017-12-02 06:04:28,308 - snapshots - ERROR - HTTPSConnectionPool(host='www.bitstamp.net', port=443): Max retries exceeded with url: /api/order_book/ (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x1188060d0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
2017-12-02 06:04:58,320 - db_util - ERROR - Failed to store in Spreads
2017-12-02 06:04:58,321 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2005, "Unknown MySQL server host 'cryptodb.ch7tzkxzans9.us-east-1.rds.amazonaws.com' (0)")
2017-12-02 10:35:40,777 - db_util - ERROR - Failed to store in Spreads
2017-12-02 10:35:40,778 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: u'INSERT INTO `Spreads` (bitstamp_ask, bitstamp_bid, bitstamp_id, gdax_ask, gdax_bid, gdax_id, time) VALUES (%s, %s, %s, %s, %s, %s, %s)'] [parameters: (u'10910.15', u'10910.00', u'10910.00', u'10962.46', u'10962.45', 4455739628, datetime.datetime(2017, 12, 2, 11, 32, 56, 689092))]
2017-12-02 10:50:13,436 - snapshots - ERROR - ====================
2017-12-02 15:37:56.867254
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-02 10:50:13,437 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Max retries exceeded with url: /products/BTC-USD/book?level=2 (Caused by SSLError(SSLError("bad handshake: SysCallError(54, 'ECONNRESET')",),))
2017-12-02 10:50:32,667 - db_util - ERROR - Failed to store in BitstampSnapshots
2017-12-02 10:50:32,667 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: 'DESCRIBE `BitstampSnapshots`']
2017-12-02 11:35:20,826 - snapshots - ERROR - ====================
2017-12-02 16:20:13.707876
Get request failed for: https://api.gdax.com/products/BTC-USD/book?level=2


2017-12-02 11:35:20,826 - snapshots - ERROR - HTTPSConnectionPool(host='api.gdax.com', port=443): Max retries exceeded with url: /products/BTC-USD/book?level=2 (Caused by SSLError(SSLError("bad handshake: SysCallError(50, 'ENETDOWN')",),))
2017-12-02 12:16:57,226 - db_util - ERROR - Failed to store in GdaxSnapshots
2017-12-02 12:16:57,261 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query') [SQL: u'INSERT INTO `GdaxSnapshots` (time, price, side, size, id) VALUES (%s, %s, %s, %s, %s)'] [parameters: ((datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10857.99', 'bid', u'6.2533949', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10854.85', 'bid', u'0.37827155', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10853.99', 'bid', u'1.274', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10853.08', 'bid', u'1.19054666', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10850.06', 'bid', u'0.011718', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10850', 'bid', u'0.01', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10848.41', 'bid', u'9.89107427', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10848.36', 'bid', u'0.60130821', '4456967013')  ... displaying 10 of 100 total bound parameter sets ...  (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10895.17', 'ask', u'1.41953164', '4456967013'), (datetime.datetime(2017, 12, 2, 17, 5, 20, 308471), u'10896', 'ask', u'0.96745965', '4456967013'))]
2017-12-02 12:18:27,297 - snapshots - ERROR - ====================
2017-12-02 17:16:57.261945
Get request failed for: https://www.bitstamp.net/api/order_book/


2017-12-02 12:18:27,298 - snapshots - ERROR - HTTPSConnectionPool(host='www.bitstamp.net', port=443): Max retries exceeded with url: /api/order_book/ (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x117fa9350>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
2017-12-02 12:18:57,316 - db_util - ERROR - Failed to store in Spreads
2017-12-02 12:18:57,316 - db_util - ERROR - (_mysql_exceptions.OperationalError) (2005, "Unknown MySQL server host 'cryptodb.ch7tzkxzans9.us-east-1.rds.amazonaws.com' (0)")
